cai_causal_graph/causal_graph.py

```diff
@@ -207,13 +209,32 @@ def adjacency_matrix(self) -> numpy.ndarray:
 
         return adjacency
 
-    @staticmethod
+    @classmethod
     def from_adjacency_matrix(
-        adjacency: numpy.ndarray, node_names: Optional[List[Union[NodeLike, int]]] = None
+        cls,
+        adjacency: numpy.ndarray,
+        node_names: Optional[List[Union[NodeLike, int]]] = None,
+        graph_class: Union[Type[CausalGraph], None] = None,
     ) -> Skeleton:
-        """Instantiate a `cai_causal_graph.causal_graph.Skeleton` object from an adjacency matrix."""
-        graph: CausalGraph = CausalGraph.from_adjacency_matrix(adjacency=adjacency, node_names=node_names)
-        return Skeleton(graph=graph)
+        """
+        Instantiate a `cai_causal_graph.causal_graph.Skeleton` object from an adjacency matrix.
+
+        :param adjacency: The adjacency matrix.
+        :param node_names: A list of the node names. These should be in the order of the rows and columns of the
+            adjacency matrix.
+        :param graph_class: The appropriate subclass of `cai_causal_graph.causal_graph.CausalGraph` to use. If `None`,
```

Comment:
@maxelliott-causalens this is the only bit where I am not sure if it is a good solution. Only affects the `from` methods for `Skeleton`.

Reply:
I don't think we can as we would want to have `TimeSeriesNode` as the node class within the `Skeleton` of a `TimeSeriesCausalGraph`.